p2p/src/network/kad/stream/p2p_network_kad_stream_effects.rs

```diff
@@ -177,6 +177,31 @@ impl P2pNetworkKademliaStreamAction {
                 });
                 Ok(())
             }
+            (
+                A::SendResponse {
+                    addr,
+                    peer_id,
+                    stream_id,
+                    data,
+                },
+                D::Incoming(I::RequestIsReady {
+                    data: P2pNetworkKademliaRpcRequest::FindNode { key },
+                }),
```

Comment:
The `P2pNetworkKademliaAction::AnswerFindNodeRequest` action dispatches before `WaitOutgoing`, so the stream ends up in the wrong state.

Reply:
Just replacing order between `WaitOutgoing` and `AnswerFindNodeRequest` fixes the issue, I completely overlooked this.

Reply:
Well, this is not an easy task to understand all the logic from the first time...